tasks/gov.js

```diff
@@ -37,31 +46,212 @@ task("gov:setPricingConfig")
     .addPositionalParam("underlying")
     .addPositionalParam("pricingType")
     .addPositionalParam("pricingParameter")
+    .addOptionalParam("isfork", "Run on localhost, which is already forked from mainnet")
     .setAction(async (args) => {
+        const isfork = args.isfork === undefined ? false : parseBool(args.isfork);
+        let fork;
+        if (isfork) {
+            fork = await setupGovernanceFork("governor");
+        }
+        const et = require("../test/lib/eTestLib");
+        const ctx = isfork ? fork.ctx: await et.getTaskCtx();
+        const underlying = isfork? args.underlying : await et.taskUtils.lookupToken(ctx, args.underlying);
+
+        if (isfork) {
+            const admin = fork.admin;
+            await et.taskUtils.runTx(ctx.contracts.governance.connect(admin).setPricingConfig(underlying, parseInt(args.pricingType), parseInt(args.pricingParameter), await ctx.txOpts()));
+        } else {
+            await et.taskUtils.runTx(ctx.contracts.governance.setPricingConfig(underlying.address, parseInt(args.pricingType), parseInt(args.pricingParameter), await ctx.txOpts()));
+        }
+    });
+
+
+task("gov:forkAccountsAndHealthScores", "Get all unique accounts that have entered an Euler market and their health scores")
+    .addPositionalParam("filename", "file name without .json suffix")
+    .setAction(async ({ filename }) => {
+        const fs = require("fs");
+
         const et = require("../test/lib/eTestLib");
-        const ctx = await et.getTaskCtx();
+        const {ctx, } = await setupGovernanceFork();
+
+        const transactions = await ctx.contracts.euler.queryFilter(
+            "EnterMarket",
+            "earliest",
+            "latest",
+        );
+
+        let result = [];
+        for (let i = 0; i < transactions.length; i++) {
+            let temp = {};
+            temp.account = transactions[i].args.account;
+            temp.underlying = transactions[i].args.underlying;
+            result.push(temp);
+        }
 
-        let underlying = await et.taskUtils.lookupToken(ctx, args.underlying);
+        // unique event arguments based on accounts
+        const key = 'account';
+        const arrayUniqueByKey = [...new Map(result.map(item =>
+            [item[key], item])).values()];
 
-        await et.taskUtils.runTx(ctx.contracts.governance.setPricingConfig(underlying.address, parseInt(args.pricingType), parseInt(args.pricingParameter), await ctx.txOpts()));
+        // unique addresses regardless of markets 
+        const uniqueAddresses = [...new Set(arrayUniqueByKey.map(item => item.account))];
+
+        // compute health scores
+        let health_scores = {};
+
+        for (let account of uniqueAddresses) {
+            let status = await ctx.contracts.exec.liquidity(account);
```

Comment:
it would be way faster if the contract was called using multicall in batches. or at least if awaiting for a batch of promises, not one by one

Reply:
Used `await Promise.all` to parse the accounts in batches and its a bit quicker. If the batch size is too large, the connection to the fork will cut out / fail with `error: Error: connect ECONNRESET 127.0.0.1:8545`

@kasperpawlowski